### DISCUSSION TITLE
Possibility to adapt for UM24C/34C USB testers

### DISCUSSION THREAD

**Discussion by hyfinn** [syssi/esphome-dps#13] [Q&A]
Hi there, thanks for your work on this. I wondered how easy it would be to adapt this code to be able to read voltage/current measurements from the UM24C/34C series of USB testers? From what I can see there are UART pins available and [well documented comms protocol](https://sigrok.org/wiki/RDTech_UM_series) - I'd love to be able to monitor DC voltage coming from my solar setup and see it in via Home Assistant.

Any ideas/thoughts would be appreciated. Thanks

**Comment by syssi** [comment]
ESPHome doesn't support serial connections via bluetooth (RFComm). If ESPHome supports RFComm some day decoding the data should be easy. May be you could sniff the serial data between the bluetooth module of the device and the microcontroller. Did you try to disassemble the device?
